pkg/k8s/api.go

```diff
@@ -126,52 +89,35 @@ func (kubeAPI *KubernetesAPI) URLFor(namespace, path string) (*url.URL, error) {
 	return generateKubernetesAPIURLFor(kubeAPI.Host, namespace, path)
 }
 
-func (kubeAPI *KubernetesAPI) getRequest(ctx context.Context, client *http.Client, path string) (*http.Response, error) {
-	endpoint, err := generateKubernetesURL(kubeAPI.Host, path)
-	if err != nil {
-		return nil, err
-	}
-
-	req, err := http.NewRequest("GET", endpoint.String(), nil)
-	if err != nil {
-		return nil, err
-	}
-
-	return client.Do(req.WithContext(ctx))
-}
-
 // NewAPI validates a Kubernetes config and returns a client for accessing the
-// configured cluster
-func NewAPI(configPath, kubeContext string) (*KubernetesAPI, error) {
+// configured cluster.
+func NewAPI(configPath, kubeContext string, timeout time.Duration) (*KubernetesAPI, error) {
```

Comment:
Not your change, but can I bug you to move this func to the top of this file, right below where the `KubernetesAPI` struct is defined? It feels weird to have the initializer func defined so far down in the file.